File: 5kyu/number-of-trailing-zeros-of-n.rs
// https://www.codewars.com/kata/number-of-trailing-zeros-of-n/train/rust

fn zeros(n: u64) -> u64 {
    let mut ans = 0u64;
    let mut d = 5;
    while d < n {
        ans += n / d;
        d *= 5;
    }
    ans
}

File: haskell/5kyu/number-of-trailing-zeros-of-n.hs
-- https://www.codewars.com/kata/number-of-trailing-zeros-of-n/train/haskell

module Zeros where

    zeros :: Int -> Int
    zeros n = zeros' 5 n
        where zeros' d n = if d < n then div n d + zeros' (d * 5) n else 0
